### PR TITLE
fix: update mapping between web component definition and JSON schema for title and description

### LINKED ISSUE #4344: Map the "title" and "description" using the same property names between the web component definition to JSON schema
# Description

The "title" and "description" mapping from the `WebComponentDefinition` and the JSON schema do not match, this should be rectified to where the name is consistent.

### PR DESCRIPTION
**Pull request by janechu**
# Description

This change more precisely matches the title and description between the web component definitions and the JSON schema.

## Motivation & context

Resolves #4344

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [x] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

**Review by EisenbergEffect — APPROVED** [review]

File: packages/tooling/fast-tooling/src/data-utilities/mapping.spec.ts [hunk 1]
```diff
@@ -1062,22 +1062,27 @@ describe("mapWebComponentDefinitionToJSONSchema", () => {
     });
     test("should map attributes and slots to the JSON schema", () => {
         const name: string = "foo";
+        const title: string = "FooBar";
         const description: string = "foo tag";
         const attrName: string = "attr";
+        const attrTitle: string = "attr title";
```
Comment (awentzel) [inline packages/tooling/fast-tooling/src/data-utilities/mapping.spec.ts:1068]: "attrName" and "attrTitle" is the only abbreviation.  Can this be spelled out for better readability and consistency?

**Review by awentzel — APPROVED** [review]

File: packages/tooling/fast-tooling/src/data-utilities/mapping.spec.ts [hunk 2]
```diff
@@ -1098,20 +1104,22 @@ describe("mapWebComponentDefinitionToJSONSchema", () => {
                 $schema: "http://json-schema.org/schema#",
                 $id: name,
                 id: name,
-                title: `<${name}>`,
+                title,
                 description: `${description}`,
```
Comment (robarbms) [inline packages/tooling/fast-tooling/src/data-utilities/mapping.spec.ts:1108]: Missed this one :)

**Review by robarbms — APPROVED** [review]

**Review by nicholasrice — APPROVED** [review]

File: packages/tooling/fast-tooling/src/data-utilities/mapping.spec.ts (hunk 2 above)
Comment (janechu) [reply packages/tooling/fast-tooling/src/data-utilities/mapping.spec.ts:1108]: Yep, haha, thanks!

```suggestion
                description,
```

File: packages/tooling/fast-tooling/src/data-utilities/mapping.spec.ts (hunk 1 above)
Comment (janechu) [reply packages/tooling/fast-tooling/src/data-utilities/mapping.spec.ts:1068]: This was present in file previously, if you'd like to create an issue for it we can put it in the backlog but I don't want to muddy the waters on the specificity for this particular change with test file var name updates.

**Pull request merged**